WordPress/openverse-frontend
# Add translation notification banner

## Fixes


Fixes #419 by @zackkrida

## Description


This PR adds a banner inviting contributors to add translations for locales where fewer than 90% of strings are translated.
The translated percentages for locales are scraped from the HTML page instead of the API due to lack of CORS header on the wp.org site.
It uses the previously created `NotificationBanner` component, but refactors it to be more flexible through using slots for any actions. The `Notification` Vuex store is also removed.

This PR replaces a 18n-sync mixin with a similar composable, and adds two more composables: use-storage and use-event-listener, mainly based on the Vueuse library code.

To add the correct link to the translation banner, I had to use a full locales list instead of the `valid-locales.json` list of locales that have at least one translated string. To make the use of both lists together easier, I refactored the locales list to use camelCase property names instead of the snake_case that was previously used.

## Screenshots
I've used the banner color that was used previously on CC Search. @panchovm , what is the best color and position for this banner? Should it be full-width as now, or have the same maximum width as the header?

<img width="1097" alt="Screen Shot 2021-12-02 at 12 37 58 PM" src="https://user-images.githubusercontent.com/15233243/144425950-0ec27890-c1b6-43ad-8d0b-e84a07fe3736.png">

<img width="1123" alt="Screen Shot 2021-12-02 at 3 15 54 PM" src="https://user-images.githubusercontent.com/15233243/144425941-fee38859-286d-48dd-af11-c761b6743140.png">


## Testing Instructions


> from Zack: You can also test this in a hacky way by editing the `onMounted` hook of use-i18n-sync with the following code: 
>   localeMsgHandler({
>     data: { type: 'localeSet', value: { locale: 'ar' } },
>   })

1. Update Nuxt locale files using `npm run i18n:create-locales-list` and then `npm run i18n:get-translations`.
2. Run the Nuxt app using `npm run dev`.
3.  Run the Openverse WordPress theme (instructions in the theme Readme).
4. Open the theme Customizer and set the embed URL to http://localhost:8443
5. Change the WP theme locale on `http://localhost:8888/wp-admin/options-general.php` options page. Note the icon next to the language select, it will help to change the language back from an unfamiliar one.
6. Reload the WP Openverse site, or open it anew by going to http://localhost:8888. You should see the Translation Banner below the header, with a correct link to the translation project on translate.wordpress.org. Note: for some reason, if I try to refresh 'http://localhost:8888/openverse', it doesn't work. Only going to the top page at 'http://localhost:8888' works.
7.  If you dismiss the banner, you should see the language code in the dismissed locales list in the localStorage. The localStorage saves the dismissed banner locale code so that if the user changes the locale, they are encourage to help with the new locale, too :)

This PR still doesn't update the translated status, which should be done regularly.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
